packages/solidstart/src/server/withServerActionInstrumentation.ts

```diff
@@ -0,0 +1,57 @@
+import { SPAN_STATUS_ERROR, handleCallbackErrors } from '@sentry/core';
+import { SEMANTIC_ATTRIBUTE_SENTRY_SOURCE, captureException, getActiveSpan, spanToJSON, startSpan } from '@sentry/node';
+import { flushIfServerless, isRedirect } from './utils';
+
+/**
+ * Wraps a server action (functions that use the 'use server' directive)
+ * function body with Sentry Error and Performance instrumentation.
+ */
+export async function withServerActionInstrumentation<A extends (...args: unknown[]) => unknown>(
+  serverActionName: string,
+  callback: A,
+): Promise<ReturnType<A>> {
+  const activeSpan = getActiveSpan();
+
+  if (activeSpan) {
+    const spanData = spanToJSON(activeSpan).data;
+
+    // In solid start, server function calls are made to `/_server` which doesn't tell us
+    // a lot. We rewrite the span's route to be that of the sever action name but only
+    // if the target is `/_server`, otherwise we'd overwrite pageloads on routes that use
+    // server actions (which are more meaningful, e.g. a request to `GET /users/5` is more
+    // meaningful than overwriting it with `GET doSomeFunctionCall`).
+    if (spanData && !spanData['http.route'] && spanData['http.target'] === '/_server') {
+      activeSpan.setAttribute('http.route', serverActionName);
+      activeSpan.setAttribute(SEMANTIC_ATTRIBUTE_SENTRY_SOURCE, 'route');
+    }
+  }
+
+  try {
+    return await startSpan(
+      {
+        op: 'function.server_action',
+        name: serverActionName,
+        attributes: {
+          [SEMANTIC_ATTRIBUTE_SENTRY_SOURCE]: 'route',
```

Comment:
is this a route though? Users could specify anything here, right? I guess `component` makes more sense? 
I tried looking up the transaction name source values but only found [this orphaned page](https://develop.sentry.dev/sdk/event-payloads/properties/transaction_info/). To me the component definition makes sense but feel free to overrule me :D

Reply:
Judging from that page, `component` also makes the most sense to me. I initially had it on `url` but changed it to `route` after @mydea looked over it. 

wdyt about using `component` instead @mydea?

Reply:
yeah, component makes sense to me too! Sorry about the misdirection to use `route` 😅

Reply:
Updated.